spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -66,12 +66,26 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
   /**
    * The cached output. So we don't compute it again when need it
    */
-  var output: B = Activity.allocate[B, T]()
+  var _output: B = Activity.allocate[B, T]()
+
+  def output: B = _output
 
   /**
    * The cached gradient of activities. So we don't compute it again when need it
    */
-  var gradInput: A = Activity.allocate[A, T]()
+  var _gradInput: A = Activity.allocate[A, T]()
+
+  def gradInput: A = _gradInput
```

Comment:
performance impact?

Reply:
the compiler should inline such getter/setter and the impact should be minimal as we also have multiple method calls within forward/backward.